csrc/id_model/indexing.cpp

```diff
@@ -399,10 +480,18 @@ Val* TensorIndexer::getLinearIndex(TensorView* tv, const Expr* expr) {
   return index;
 }
 
+// Get the loop domains of a given expr, which are (potentially
+// promoted) loop domains of the consumer tensor.
 std::vector<IterDomain*> TensorIndexer::getLoopDomains(const Expr* expr) const {
   // Assume consumer-based indexing. Needs to revisit for ops like
   // scatter
-  return ir_utils::getTvOutput(expr)->getLeafDomain();
+  auto loop_domains = ir_utils::getTvOutput(expr)->getLeafDomain();
```

Comment:
Another place where we assume all outputs have same domain; in this case leaf domain.

Reply:
Yes. I think lifting that restriction is quite challenging. We would need to change many things, including expression sorting, loop generation etc.